charts/rancher-turtles/templates/deployment.yaml

```diff
@@ -26,6 +26,7 @@ spec:
       containers:
       - args:
         - --leader-elect
+        - --feature-gates=rancher-kube-secret-patch={{ index .Values "rancherTurtles" "features" "rancher-kubeconfigs" "label"}}
```

Comment:
I see, it is configurable, my mistake. But still, this could be always enabled, why would we it be disabled even if the patch merges into rancher? We should just remove this code afterwards.

Reply:
I did consider something like [OpenFeature](https://openfeature.dev/) but went with the upstream approach initially. We could consider open feature in the future.

Reply:
The implementation is behind a feature gate because when the functinality is merged into Rancher we have the option of disabling this feature.

Reply:
And what about removing this completely *after* it *finally* gets merged to rancher? This duplication puzzles me. Upstream does not use helm, and I was referring to the usage through the templating, not the use of FG source code from kubernetes. What I meant by configurability - we may have more FG like that in the future. There could be a list of values `FGA=true,FGB=false` here. Right now it is not a feature affecting some functionality or changing existing, it is a required code change to support CAPI v1.5+. I think this should be hidden on helm templates, and enabled by default in the code at this point.

Reply:
You could argue the same for the cleanup of the webhooks. I don't see a problem in exposing it....at least there is no "magic" hidden from the users.

Reply:
+1 to what @richardcase says, I prefer the upstream approach too.